extensions/github1s/src/providers/fileSystemProvider/index.ts

```diff
@@ -246,7 +246,12 @@ export class GitHub1sFileSystemProvider
 			}
 			return parent.getNameTypePairs();
 		},
-		(uri) => uri.toString()
+		/**
+		 * Since we either use the uri.authority or call `router.getAuthority()` to get the authority,
+		 * And it uses the query to distinguish between files in different branch for diff view. It's safe
+		 * to ignore authority to deduplicate the redundant network calls.
+		 */
+		(uri) => uri.with({ authority: '' }).toString()
```

Comment:
Thank you so much for finding the reason of this problem!

I think this solution is safe now, but I'm afraid it's potentially dangerous. (when we read two directories with different authority at the same time)
How do you think the solution below:
~~~typescript
async (uri) => (uri.authority ? uri : uri.with({ authority: await router.getAuthority() })).toString()
~~~

Of course, we have to modify the code of `reuseable`:

~~~typescript
export const reuseable = <A extends any[], R>(
	func: (...args: A) => Promise<R>,
	computeCacheKey: (
		...args: A
	) => string | Promise<string> = defaultComputeCacheKey
) => {
	const cache = new Map<string, Promise<R>>();

	return async function (...args: A): Promise<R> {
		const key = await computeCacheKey(...args);
		if (cache.has(key)) {
			return cache.get(key);
		}

		const promise = func.call(this, ...args);
		cache.set(key, promise);
		return pFinally(promise, () => cache.delete(key));
	};
};
~~~

I have changed the type definition of `reuseable` because of the contravariance, I don't know if there is a more elegant solution...

Reply:
I think I found a better way to do this, let's discuss it in https://github.com/conwnet/github1s/pull/247